### PR TITLE
Inject property values in autonomic "measures" files

### LINKED ISSUE #157: Make "measures" file with injectable values
Currently, **measures** files contain rules with some values (numeric, most of the time).
It would be nice to be able to edit these files through a web interface. This implies we should be able to update **measures** files from the DM.

Example of updated file:

```
[EVENT parserId eventId PARAM1 PARAM2...]
Query... value > PARAM1
...
```

PARAM1 references a parameter.
There should be another file to describe these parameters.
It should be called _component name_**.measures.properties**.

Such a file would define the properties of parameters.

``` properties
PARAM1.description = My parameter 1
PARAM1.default = 50
PARAM1.value = 

# ...
```

When a value changes, only the **measures.properties** file would be updated from the DM.
This file being on both the agent and the DM, we could also use it in the web interface to edit its values.

### PR DESCRIPTION
**Pull request by gibello**
Closes #157 issue.

### CODE REVIEW

File: core/roboconf-agent-monitoring/src/main/java/net/roboconf/agent/monitoring/internal/MonitoringTask.java [hunk 1]
```diff
@@ -103,9 +105,24 @@ public void run() {
...
+			if(paramFile.exists()) {
+				this.logger.fine( "A file with measure parameters (properties) was found for instance '" + inst + "'." );
+				FileInputStream in = null;
+				try {
+					params = new Properties();
+					params.load((in = new FileInputStream(paramFile)));
+				} catch(IOException e) {
+					params = null;
+				} finally {
+					if(in != null) try { in.close(); } catch(Exception e) { /*ignore*/ }
```
Comment (vincent-zurczak) [inline core/roboconf-agent-monitoring/src/main/java/net/roboconf/agent/monitoring/internal/MonitoringTask.java]: Use Utils.closeQuietly( in ) instead.

File: core/roboconf-agent-monitoring/src/main/java/net/roboconf/agent/monitoring/internal/MonitoringTask.java [hunk 2]
```diff
@@ -138,14 +155,14 @@ public void run() {
 				continue;
 
 			if( s.toLowerCase().startsWith( RULE_BEGINNING )) {
-				sections.add( sb.toString());
+				if(sb.length() > 0) sections.add( sb.toString());
```
Comment (vincent-zurczak) [inline core/roboconf-agent-monitoring/src/main/java/net/roboconf/agent/monitoring/internal/MonitoringTask.java]: Is this really usefl to check if sb.length() is positive?
You append "\n" below, so, even if expandString return "", you will add a new section anyway.

File: core/roboconf-core/src/main/java/net/roboconf/core/Constants.java [hunk 3]
```diff
@@ -85,7 +85,7 @@
 	/**
 	 * The <strong>rules</strong> file extension.
 	 */
-	String FILE_EXT_RULES = ".rules";
+	String FILE_RULES = "rules.cfg";
```
Comment (vincent-zurczak) [inline core/roboconf-core/src/main/java/net/roboconf/core/Constants.java]: Why do we need the *.cfg extension?
Measures files have a *.measures extension.

File: core/roboconf-dm/src/main/java/net/roboconf/dm/internal/autonomic/RulesParser.java [hunk 4]
```diff
@@ -67,8 +67,7 @@ private RulesParser() {
 	public static Map<String,AutonomicRule> parseRules( ManagedApplication ma ) throws IOException {
 
 		Map<String,AutonomicRule> result = new HashMap<String,AutonomicRule> ();
-		String fileName = ma.getApplication().getName() + Constants.FILE_EXT_RULES;
```
Comment (vincent-zurczak) [inline core/roboconf-dm/src/main/java/net/roboconf/dm/internal/autonomic/RulesParser.java]: Compare with the code of ResourceUtils.java, line 70.
You do not look for the same file... In one case, it is **component-name.rules.cfg**, and in the other, it is **rules.cfg**.

File: core/roboconf-agent-monitoring/src/main/java/net/roboconf/agent/monitoring/internal/MonitoringTask.java (hunk 1 above)
Comment (gibello) [inline core/roboconf-agent-monitoring/src/main/java/net/roboconf/agent/monitoring/internal/MonitoringTask.java]: Fixed in latest commit.

File: core/roboconf-agent-monitoring/src/main/java/net/roboconf/agent/monitoring/internal/MonitoringTask.java (hunk 2 above)
Comment (gibello) [inline core/roboconf-agent-monitoring/src/main/java/net/roboconf/agent/monitoring/internal/MonitoringTask.java]: The check was wrong - fixed now (not to include empty sections). Not so important (empty sections are filtered out when present...) but right !

**Pull request merged**